contracts/Announcer.sol

```diff
@@ -5,10 +5,12 @@ import "./IAnnounce.sol";
 
 contract Announcer is IAnnounce {
     /**
-     * @param hash Keccak256 hash of the content found at uri
-     * @param dsnpUri string URL, ipfs, etc... to the content
+     * @param announcements array of struct announcements
      */
-    function batch(bytes32 hash, string calldata dsnpUri) external override {
-        emit DSNPBatch(hash, dsnpUri);
+    function batch(Announcement[] calldata announcements) external override {
+        require(announcements.length < 1000, "gas consumption is high");
+        for (uint256 i = 0; i < announcements.length; i++) {
```

Comment:
do we want any limit checking on the dsnpType value, that is, do we care if they announce a nonsense type? I think we might since it's indexed.  Unfortunately that would also mean updating some contract whenever new types were added.

Reply:
Yeah. Thats a good question, perhaps we can discuss this tomorrow.